ci/generate_and_upload_build_steps.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script generates steps for each engine version listed in unreal-engine.version, and adds those to generated_base.steps.yaml
+# The steps are based on the template in generated_steps.steps.yaml
+
+if [ -z "${ENGINE_VERSION}" ]; then 
+    echo "Generating build steps for each engine version listed in unreal-engine.version"  
+    IFS=$'\n'
```

Comment:
For my own knowledge, could you explain how this for loop knows how to delimit via the newline

Reply:
Sure thing. In bash `IFS` is a special "internal field separator" variable. That variable is used in, among other places, the `for` construct to determine what to use as delimiters. In our case, `IFS` is set to a newline character on line 9 (IFS=$'\n').